soroban-sdk/src/deploy.rs

```diff
@@ -258,38 +260,36 @@ impl DeployerWithAddress {
     /// and provided salt.
     ///
     /// Returns the deployed contract's address.
+    #[deprecated(note = "use deploy_v2")]
     pub fn deploy(&self, wasm_hash: impl IntoVal<Env, BytesN<32>>) -> Address {
-        let env = &self.env;
-        let address_obj = env
-            .create_contract(
-                self.address.to_object(),
-                wasm_hash.into_val(env).to_object(),
-                self.salt.to_object(),
-            )
-            .unwrap_infallible();
-        unsafe { Address::unchecked_new(env.clone(), address_obj) }
+        self.deploy_v2(wasm_hash, ())
```

Comment:
Undo this change, so that deploy v1 still calls the old host fn.
```suggestion
        let env = &self.env;
        let address_obj = env
            .create_contract(
                self.address.to_object(),
                wasm_hash.into_val(env).to_object(),
                self.salt.to_object(),
            )
            .unwrap_infallible();
        unsafe { Address::unchecked_new(env.clone(), address_obj) }
```